src/libs/ReportUtils.ts

```diff
@@ -739,13 +739,17 @@ const unavailableTranslation = Localize.translateLocal('workspace.common.unavail
  */
 function getPolicyName(report: OnyxInputOrEntry<Report>, returnEmptyIfNotFound = false, policy?: OnyxInputOrEntry<Policy>): string {
     const noPolicyFound = returnEmptyIfNotFound ? '' : unavailableTranslation;
-    if (isEmptyObject(report)) {
+    
+    // Ensure policies and report are fully loaded before proceeding.
+    // On initial login, they may be partially loaded, so additional checks are necessary to prevent unexpected behavior.
+    if (isEmptyObject(report) || !report.reportID || isEmptyObject(allPolicies)) {
         return noPolicyFound;
     }
-
-    if ((!allPolicies || Object.keys(allPolicies).length === 0) && !report?.policyName) {
+    
+    if (isEmptyObject(allPolicies) && !report?.policyName) {
```

Comment:
@allgandalf upon further consideration, I believe the second check is unreachable. My suggestion would be to eliminate it. Your thoughts?